gips/data/hls/hls.py

```diff
@@ -150,13 +166,95 @@ def query_provider(cls, asset, tile, date, **ignored):
         if requests.head(url).status_code == 200: # so do they have it?
             return basename, url
-        return None, None
+        return None
+
+    _s3_bucket_name = "hlsp1"
+    _s3_base_key = "HLS_LOCAL_IO.v1.4"
 
     @classmethod
-    def download(cls, url, download_fp, pclouds=100.0, **ignored):
+    def query_s3(cls, asset, tile, date, **ignored):
+
+        # TODO: check version in S3 mode
+        #cls.check_hls_version()
+
+        # build the full URL & basename of the file
+        basename = 'HLS.{}.T{}.{}.v{}.hdf'.format(
+            asset, tile, date.strftime('%Y%j'), _hls_version)
+
+        s30_key = '{base}/S30/{year}/{tile1}/{tile2}/{tile3}/{tile4}/'\
+                  'HLS.S30.T{tile}.{datestr}.v{version}.hdf'\
+                  .format(base=cls._s3_base_key, year=date.year, tile1=tile[0:2],
+                          tile2=tile[2], tile3=tile[3], tile4=tile[4],
+                          tile=tile, datestr=date.strftime('%Y%j'),
+                          version='1.4')
+
+        l30_key = '{base}/L30/{year}/{tile1}/{tile2}/{tile3}/{tile4}/'\
+                  'HLS.S30.T{tile}.{datestr}.v{version}.hdf'\
+                  .format(base=cls._s3_base_key, year=date.year, tile1=tile[0:2],
+                          tile2=tile[2], tile3=tile[3], tile4=tile[4],
+                          tile=tile, datestr=date.strftime('%Y%j'),
+                          version='1.4')
+
+        # the user must have a [nasa] profile in their $HOME/.aws/credentials
+        s30keys = cls.s3_prefix_search(s30_key, profile='nasa')
+        l30keys = cls.s3_prefix_search(l30_key, profile='nasa')
+
+        if len(s30keys) > 0:
+            key = [k for k in s30keys if '.hdf' in k][0]
+        elif len(l30keys) > 0:
+            key = [k for k in l30keys if '.hdf' in k][0]
+        else:
+            return None
+
+        url = 's3://{bucket}/{key}'.format(bucket=cls._s3_bucket_name, key=key)
+        return basename, url
+
+    @classmethod
+    @lru_cache(maxsize=100) # cache size chosen arbitrarily
+    def query_service(cls, asset, tile, date, **ignored):
+        # start with pre-query checks
+        if not cls.available(asset, date):
+            return None
+        if cls.data_src() == 'http':
+            result = cls.query_http(asset, tile, date)
+        else:
+            result = cls.query_s3(asset, tile, date)
+        # caller needs either None or this dictionary
+        if result is not None:
+            bn, url = result
+            result = {'basename': bn, 'url': url}
+        return result
+
+    @classmethod
+    def download_http(cls, url, download_fp, pclouds=100.0):
         utils.http_download(url, download_fp)
+        print(url, download_fp)
         return cls(download_fp).filter(pclouds)
 
+    @classmethod
+    def download_s3(cls, url, download_fp, pclouds=100.0):
+        import boto3
+        boto3.setup_default_session(profile_name='nasa')
+        s3_client = boto3.client('s3')
```

Comment:
If this is being hardcoded to always use our Indigo NASA credentials, then it won't work for the general population who uses GIPS.  Or you should be more clear with a comment or a log warning that the 'nasa' profile name needs to be parameterized somewhere and linked with an AWS S3 key.

Reply:
I left the capability for a driver to specify a profile in AWS credentials file, which I think would work in general as you say as long as the host knows about it. Just for now I have a workaround that is not great but is a little more general.